server/rest/src/main/java/org/infinispan/rest/resources/ClusterResource.java

```diff
@@ -107,33 +116,67 @@ private CompletionStage<RestResponse> restore(RestRequest request) {
    }
 
    private CompletionStage<RestResponse> distribution(RestRequest request) {
-      return clusterDistribution()
+      return requestCluster(ecm -> NodeDistributionInfo.resolve(ecm.getCacheManagerInfo(), SecurityActions.getGlobalComponentRegistry(ecm)))
             .thenApply(distributions -> asJsonResponse(Json.array(distributions.stream()
                   .map(NodeDistributionInfo::toJson).toArray())));
    }
 
-   private CompletionStage<List<NodeDistributionInfo>> clusterDistribution() {
+   private CompletionStage<RestResponse> connectors(RestRequest request) {
+      String qualifiedRestName = invocationHelper.getProtocolServer().getQualifiedName();
```

Comment:
My main concern here is the REST connector configured with different names across the cluster. In such cases, the component would not be found, and the info is not returned.

Reply:
Does not apply anymore.